protocols/imap/src/test/java/org/apache/james/imap/encode/AnnotationResponseEncoderTest.java

```diff
@@ -19,70 +19,75 @@
+    }
+
     @Test
     public void annotationMessageResponseShouldBeWellFormattedWithoutReturnAnnotation() throws Exception {
         AnnotationResponse response = new AnnotationResponse("INBOX", ImmutableList.<MailboxAnnotation>of());
-        ByteImapResponseWriter byteImapResponseWriter = new ByteImapResponseWriter();
-        ImapResponseComposer composer = new ImapResponseComposerImpl(byteImapResponseWriter, 1024);
-        AnnotationResponseEncoder encoder = new AnnotationResponseEncoder(new EndImapEncoder());
-        encoder.encode(response, composer, null);
-        String responseString = byteImapResponseWriter.getString();
 
-        assertEquals("* METADATA \"INBOX\"\r\n", responseString);
+        encoder.encode(response, composer, imapSession);
+
+        assertThat(byteImapResponseWriter.getString()).isEqualTo("* METADATA \"INBOX\"\r\n");
     }
 
     @Test
     public void annotationMessageResponseShouldBeWellFormattedWithOneReturnAnnotation() throws Exception {
         AnnotationResponse response = new AnnotationResponse("INBOX", ImmutableList.of(PRIVATE_ANNOTATION));
-        ByteImapResponseWriter byteImapResponseWriter = new ByteImapResponseWriter();
-        ImapResponseComposer composer = new ImapResponseComposerImpl(byteImapResponseWriter, 1024);
-        AnnotationResponseEncoder encoder = new AnnotationResponseEncoder(new EndImapEncoder());
-        encoder.encode(response, composer, null);
-        String responseString = byteImapResponseWriter.getString();
 
-        assertEquals("* METADATA \"INBOX\" (/private/comment \"My own comment\")\r\n", responseString);
+        encoder.encode(response, composer, imapSession);
+
+        assertThat(byteImapResponseWriter.getString()).isEqualTo("* METADATA \"INBOX\" (/private/comment \"My own comment\")\r\n");
     }
 
     @Test
     public void annotationMessageResponseShouldBeWellFormattedWithManyReturnAnnotation() throws Exception {
         AnnotationResponse response = new AnnotationResponse("INBOX", ImmutableList.of(PRIVATE_ANNOTATION, SHARED_ANNOTATION));
-        ByteImapResponseWriter byteImapResponseWriter = new ByteImapResponseWriter();
-        ImapResponseComposer composer = new ImapResponseComposerImpl(byteImapResponseWriter, 1024);
-        AnnotationResponseEncoder encoder = new AnnotationResponseEncoder(new EndImapEncoder());
-        encoder.encode(response, composer, null);
-        String responseString = byteImapResponseWriter.getString();
+        encoder.encode(response, composer, imapSession);
 
-        assertEquals("* METADATA \"INBOX\" (/private/comment \"My own comment\" /shared/comment \"Shared comment\")\r\n", responseString);
+        assertThat(byteImapResponseWriter.getString()).isEqualTo("* METADATA \"INBOX\" (/private/comment \"My own comment\" /shared/comment \"Shared comment\")\r\n");
     }
 
     @Test
     public void annotationMessageResponseShouldBeWellFormattedWithNilAnnotation() throws Exception {
```

Comment:
Is it normal (/ an expected behaviour ?) to return NIL ?
I guess we are not supposed to store NIL annotations, maybe we should avoid responding them...
I propose logging at warn level + doing nothing if it is Nil...

And you need a test showing you do not answer NIL annotation